File: packages/archived/sdk-ts/src/resources/agents.ts
// @ts-nocheck
/**
 * Agents Orchestration API
 */

import type {
  Agent,
  AgentOrchestrationParams,
  AgentOrchestrationResponse,
  AgentType,
} from '../types/index.js';
import type { ClaudeFlareClient } from '../client.js';
import { errorFromResponse } from '../utils/errors.js';
import { getLogger } from '../utils/logger.js';

const logger = getLogger();

/**
 * Agent Orchestration Resource
 */
export class AgentOrchestration {
  constructor(private client: ClaudeFlareClient) {}

  /**
   * Orchestrate agents for a task
   */
  async create(params: AgentOrchestrationParams): Promise<AgentOrchestrationResponse> {
    const url = `/${this.client.config.apiVersion}/agents/orchestrate`;
    const requestOptions = this.buildRequestOptions(params);

    logger.debug('Orchestrating agents', { params, url });

    const startTime = Date.now();

    try {
      const response = await this.client.request('POST', url, requestOptions);

      const duration = Date.now() - startTime;
      logger.logResponse(response.status, duration);

      const data = await response.json();

      if (!response.ok) {
        throw errorFromResponse(response.status, data, this.client.getRequestId(response));
      }

      return data;
    } catch (error) {
      logger.error('Agent orchestration failed', { error, duration: Date.now() - startTime });
      throw error;
    }
  }

  /**
   * Orchestrate with streaming updates
   */
  async createStream(
    params: AgentOrchestrationParams,
    onUpdate: (update: AgentOrchestrationResponse) => void
  ): Promise<AgentOrchestrationResponse> {
    const url = `/${this.client.config.apiVersion}/agents/orchestrate`;
    const streamParams = { ...params };
    const requestOptions = this.buildRequestOptions(streamParams);

    logger.debug('Orchestrating agents with streaming', { params: streamParams, url });

    const startTime = Date.now();

    try {
      const response = await this.client.request('POST', url, {
        ...requestOptions,
        headers: {
          ...requestOptions.headers,
          Accept: 'text/event-stream',
        },
      });

      if (!response.ok) {
        const data = await response.json();
        throw errorFromResponse(response.status, data, this.client.getRequestId(response));
      }

      const duration = Date.now() - startTime;
      logger.logResponse(response.status, duration);

      // Process stream
      const reader = response.body?.getReader();
      const decoder = new TextDecoder();
      let finalResponse: AgentOrchestrationResponse | null = null;

      if (reader) {
        while (true) {
          const { done, value } = await reader.read();

          if (done) break;

          const chunk = decoder.decode(value);
          const lines = chunk.split('\n');

          for (const line of lines) {
            if (line.startsWith('data: ')) {
              const data = line.slice(6);

              if (data === '[DONE]') {
                break;
              }

              try {
                const update = JSON.parse(data) as AgentOrchestrationResponse;
                onUpdate(update);

                if (update.status === 'completed' || update.status === 'failed') {
                  finalResponse = update;
                }
              } catch {
                // Skip invalid JSON
              }
            }
          }
        }
      }

      if (!finalResponse) {
        throw new Error('Orchestration did not complete');
      }

      return finalResponse;
    } catch (error) {
      logger.error('Agent orchestration streaming failed', { error, duration: Date.now() - startTime });
      throw error;
    }
  }

  /**
   * Build request options
   */
  private buildRequestOptions(params: AgentOrchestrationParams): RequestInit {
    return {
      body: JSON.stringify({
        task: params.task,
        agents: params.agents,
        auto_select: params.autoSelect,
        context: params.context,
        max_parallelism: params.maxParallelism,
        timeout: params.timeout,
      }),
    };
  }
}

/**
 * Agent Registry Resource
 */
export class AgentRegistry {
  constructor(private client: ClaudeFlareClient) {}

  /**
   * Get agent registry status
   */
  async getStatus(): Promise<{
    status: string;
    agents: Agent[];
    timestamp: number;
  }> {
    const url = `/${this.client.config.apiVersion}/agents/status`;

    logger.debug('Getting agent registry status', { url });

    const startTime = Date.now();

    try {
      const response = await this.client.request('GET', url);

      const duration = Date.now() - startTime;
      logger.logResponse(response.status, duration);

      const data = await response.json();

      if (!response.ok) {
        throw errorFromResponse(response.status, data, this.client.getRequestId(response));
      }

      return data;
    } catch (error) {
      logger.error('Failed to get agent registry status', { error, duration: Date.now() - startTime });
      throw error;
    }
  }

  /**
   * Get available agents
   */
  async list(type?: AgentType): Promise<{
    agents: Agent[];
    count: number;
    timestamp: number;
  }> {
    const url = type
      ? `/${this.client.config.apiVersion}/agents/available/${type}`
      : `/${this.client.config.apiVersion}/agents/available`;

    logger.debug('Listing available agents', { url, type });

    const startTime = Date.now();

    try {
      const response = await this.client.request('GET', url);

      const duration = Date.now() - startTime;
      logger.logResponse(response.status, duration);

      const data = await response.json();

      if (!response.ok) {
        throw errorFromResponse(response.status, data, this.client.getRequestId(response));
      }

      return data;
    } catch (error) {
      logger.error('Failed to list available agents', { error, duration: Date.now() - startTime });
      throw error;
    }
  }

  /**
   * Get agents by type
   */
  async getByType(type: AgentType): Promise<Agent[]> {
    const result = await this.list(type);
    return result.agents;
  }

  /**
   * Get all available agents
   */
  async getAll(): Promise<Agent[]> {
    const result = await this.list();
    return result.agents;
  }
}

/**
 * Agents API namespace
 */
export class Agents {
  constructor(
    public orchestrate: AgentOrchestration,
    public registry: AgentRegistry
  ) {}
}

File: packages/archived/sdk-python/src/claudeflare/resources/agents.py
"""
Agent orchestration API
"""

import logging
from typing import Any, AsyncIterator

from claudeflare.types import (
    AgentOrchestrationParams,
    AgentOrchestrationResponse,
    Agent,
    AgentType,
)
from claudeflare.client import ClaudeFlare
from claudeflare.exceptions import error_from_response

logger = logging.getLogger("claudeflare")


class AgentOrchestration:
    """Agent orchestration resource."""

    def __init__(self, client: ClaudeFlare):
        self.client = client

    async def create(self, params: AgentOrchestrationParams) -> AgentOrchestrationResponse:
        """
        Orchestrate agents for a task.

        Args:
            params: Orchestration parameters

        Returns:
            Orchestration response
        """
        endpoint = "agents/orchestrate"
        body = {
            "task": params.task,
            "agents": [a.value for a in params.agents] if params.agents else None,
            "auto_select": params.auto_select,
            "context": params.context,
            "max_parallelism": params.max_parallelism,
            "timeout": params.timeout,
        }

        body = {k: v for k, v in body.items() if v is not None}

        logger.debug(f"Orchestrating agents: {endpoint}")

        response = await self.client.post(endpoint, json_data=body)

        if not response.is_error:
            data = response.json()
            return AgentOrchestrationResponse(**data)

        error = error_from_response(response.status_code, response.json())
        raise error

    async def create_stream(
        self, params: AgentOrchestrationParams
    ) -> AsyncIterator[AgentOrchestrationResponse]:
        """
        Orchestrate agents with streaming updates.

        Args:
            params: Orchestration parameters

        Yields:
            Orchestration updates
        """
        endpoint = "agents/orchestrate"
        body = {
            "task": params.task,
            "agents": [a.value for a in params.agents] if params.agents else None,
            "auto_select": params.auto_select,
            "context": params.context,
            "max_parallelism": params.max_parallelism,
            "timeout": params.timeout,
        }

        body = {k: v for k, v in body.items() if v is not None}

        logger.debug(f"Orchestrating agents with streaming: {endpoint}")

        response = await self.client.post(
            endpoint,
            json_data=body,
            headers={"Accept": "text/event-stream"},
        )

        if response.is_error:
            error = error_from_response(response.status_code, response.json())
            raise error

        async for line in response.aiter_lines():
            if not line.startswith("data: "):
                continue

            data = line[6:]

            if data == "[DONE]":
                break

            try:
                import json

                update = json.loads(data)
                yield AgentOrchestrationResponse(**update)
            except json.JSONDecodeError:
                logger.debug(f"Failed to parse update: {data}")


class AgentRegistry:
    """Agent registry resource."""

    def __init__(self, client: ClaudeFlare):
        self.client = client

    async def get_status(self) -> dict[str, Any]:
        """Get agent registry status."""
        endpoint = "agents/status"

        logger.debug(f"Getting agent registry status: {endpoint}")

        response = await self.client.get(endpoint)

        if not response.is_error:
            return response.json()

        error = error_from_response(response.status_code, response.json())
        raise error

    async def list(self, agent_type: AgentType | None = None) -> dict[str, Any]:
        """
        List available agents.

        Args:
            agent_type: Filter by agent type

        Returns:
            List of agents
        """
        endpoint = f"agents/available/{agent_type.value}" if agent_type else "agents/available"

        logger.debug(f"Listing available agents: {endpoint}")

        response = await self.client.get(endpoint)

        if not response.is_error:
            return response.json()

        error = error_from_response(response.status_code, response.json())
        raise error

    async def get_all(self) -> list[Agent]:
        """Get all available agents."""
        result = await self.list()
        return [Agent(**a) for a in result.get("agents", [])]

    async def get_by_type(self, agent_type: AgentType) -> list[Agent]:
        """Get agents by type."""
        result = await self.list(agent_type)
        return [Agent(**a) for a in result.get("agents", [])]


class Agents:
    """Agents API namespace."""

    def __init__(self, orchestrate: AgentOrchestration, registry: AgentRegistry):
        self.orchestrate = orchestrate
        self.registry = registry
